### PR TITLE
Update README.md

### PR DESCRIPTION
**Pull request by supriyar**
Updates to include 

1. the interoperability details
2. our goals
3. some general cleanup

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -132,6 +156,13 @@ smooth_fq_linear_to_inference(model)
 model = torch.compile(model, mode='max-autotune')
 model(input)
 ```
+## Success stories
+Our kernels have has been used to achieve SOTA inference performance on
+
+1. Image segmentation modelss with [sam-fast](pytorch.org/blog/accelerating-generative-ai)
```
Comment (msaroufim) [inline README.md:162]:
```suggestion
1. Image segmentation models with [sam-fast](pytorch.org/blog/accelerating-generative-ai)
```

File: README.md [hunk 2]
```diff
@@ -1,20 +1,30 @@
-# torchao: PyTorch Architecture Optimization
```
Comment (msaroufim) [inline README.md:1]: first question i always get is what does ao mean so let's answer that somewheere in the first 3 lines

File: README.md [hunk 3]
```diff
@@ -1,20 +1,30 @@
...
-3. Sparsity [algorithms](./torchao/sparsity) such as Wanda
+torchao is a PyTorch native library for optimizing your models using lower precision dtypes, techniques like quantization and sparsity and performant kernels.
 
-## Success stories
-Our kernels have has been used to achieve SOTA inference performance on
+The library provides
+1. Support for lower precision [dtypes](./torchao/dtypes) such as nf4, uint4 that are torch.compile friendly
+2. Quantization [algorithms](./torchao/quantization) such as dynamic quant, smoothquant, GPTQ that run on CPU/GPU and Mobile.
+3. Sparsity [algorithms](./torchao/sparsity) such as Wanda that help improve accuracy of sparse networks
+4. Integration with other PyTorch native libraries like TorchTune and ExecuTorch
```
Comment (msaroufim) [inline README.md:11]:
```suggestion
4. Integration with other PyTorch native libraries like torchtune and ExecuTorch
```

File: README.md [hunk 4]
```diff
@@ -1,20 +1,30 @@
...
 
-1. Image segmentation modelss with [sam-fast](pytorch.org/blog/accelerating-generative-ai)
-2. Language models with [gpt-fast](pytorch.org/blog/accelerating-generative-ai-2)
-3. Diffusion models with [sd-fast](pytorch.org/blog/accelerating-generative-ai-3)
+## Key Features
+* Native PyTorch techniques, composable with torch.compile
+* APIs tested on following hardware - A100, T4(colab)
```
Comment (msaroufim) [inline README.md:15]: this is not really a feature maybe just put somewhere near the bottom - hardware tested on A100 and T4

File: README.md [hunk 5]
```diff
@@ -1,20 +1,30 @@
...
+  * Int8 dynamic activation quantization with Smoothquant
+  * Int8 and int4 weight-only quantization.
+  * Int8 dynamic activation quantization with int4 weight quantization
+  * GPTQ
+
+## Interoperability with PyTorch Libraries
+
+torchao has been integrated with other repositories to ease usage
+
+* [TorchTune](https://github.com/pytorch/torchtune/blob/main/recipes/quantization.md) is integrated with 8 and 4 bit weight-only quantization techniques with and without GPTQ.
```
Comment (msaroufim) [inline README.md:26]:
```suggestion
* [torchtune](https://github.com/pytorch/torchtune/blob/main/recipes/quantization.md) is integrated with 8 and 4 bit weight-only quantization techniques with and without GPTQ.
```

File: README.md [hunk 6]
```diff
@@ -1,20 +1,30 @@
...
+3. Sparsity [algorithms](./torchao/sparsity) such as Wanda that help improve accuracy of sparse networks
+4. Integration with other PyTorch native libraries like TorchTune and ExecuTorch
 
-1. Image segmentation modelss with [sam-fast](pytorch.org/blog/accelerating-generative-ai)
-2. Language models with [gpt-fast](pytorch.org/blog/accelerating-generative-ai-2)
-3. Diffusion models with [sd-fast](pytorch.org/blog/accelerating-generative-ai-3)
+## Key Features
+* Native PyTorch techniques, composable with torch.compile
+* APIs tested on following hardware - A100, T4(colab) 
+* Supports the following eager quantization techniques.
```
Comment (msaroufim) [inline README.md:16]: so which techniques are compile specific and which are eager? If it's the same set then we can set say something like kernels that work with both eager and compile such as: int8 etc..

File: README.md (hunk 2 above)
Comment (supriyar) [reply README.md:1]: I think this got deleted by mistake, I'll add it back

File: README.md [hunk 7]
```diff
@@ -132,6 +156,13 @@ smooth_fq_linear_to_inference(model)
 model = torch.compile(model, mode='max-autotune')
 model(input)
 ```
+## Success stories
```
Comment (msaroufim) [inline README.md:159]: I'd move these up near the top they're more important than features and a lot of people know the fast series already

File: README.md [hunk 8]
```diff
@@ -34,6 +44,20 @@ cd ao
...
+## Our Goals
+torchao embodies PyTorch’s design philosophy [details](https://pytorch.org/docs/stable/community/design.html), especially "usability over everything else". Our vision for this repository is the following:
+
+* PyTorch Native: Provide PyTorch native solutions for optimization techniques for transformer models, offering composability with torch.compile and distributed 
+    * For example, for QLoRA for new dtypes support
+* Interoperability: Integration with other major transformer accel libraries in PyTorch enabling smooth user journey and accelerating visibility and adoption
+    * Examples - TorchTune, gpt-fast, ExecuTorch.
+* Transparent Benchmarks: Regularly run performance benchmarking of our APIs across a suite of Torchbench models and across hardware backends
+* Heterogeneous Hardware: Efficient kernels that can run on CPU/GPU based server (w/ torch.compile) and mobile backends (w/ ExecuTorch).
+* Autotuner: Invest in building a kernel auto tuner targeting SOTA performance across varying model shapes.
```
Comment (msaroufim) [inline README.md:56]: i'd move the autotuner to the features section as in it helps varying shapes and also both consumer and enterprise gpus

File: README.md [hunk 9]
```diff
@@ -34,6 +44,20 @@ cd ao
 pip install -e .
 ```
 
+## Our Goals
+torchao embodies PyTorch’s design philosophy [details](https://pytorch.org/docs/stable/community/design.html), especially "usability over everything else". Our vision for this repository is the following:
+
+* PyTorch Native: Provide PyTorch native solutions for optimization techniques for transformer models, offering composability with torch.compile and distributed 
+    * For example, for QLoRA for new dtypes support
+* Interoperability: Integration with other major transformer accel libraries in PyTorch enabling smooth user journey and accelerating visibility and adoption
+    * Examples - TorchTune, gpt-fast, ExecuTorch.
```
Comment (msaroufim) [inline README.md:53]:
```suggestion
    * Examples - torchtune, gpt-fast, ExecuTorch.
```

File: README.md [hunk 10]
```diff
@@ -34,6 +44,20 @@ cd ao
 pip install -e .
 ```
 
+## Our Goals
+torchao embodies PyTorch’s design philosophy [details](https://pytorch.org/docs/stable/community/design.html), especially "usability over everything else". Our vision for this repository is the following:
+
+* PyTorch Native: Provide PyTorch native solutions for optimization techniques for transformer models, offering composability with torch.compile and distributed
```
Comment (msaroufim) [inline README.md:50]: Reframe as composability
```suggestion
* Composability: Native solutions for optimization techniques that compose with both `torch.compile` and `FSDP`
```

File: README.md [hunk 11]
```diff
@@ -34,6 +44,20 @@ cd ao
 pip install -e .
 ```
 
+## Our Goals
+torchao embodies PyTorch’s design philosophy [details](https://pytorch.org/docs/stable/community/design.html), especially "usability over everything else". Our vision for this repository is the following:
+
+* PyTorch Native: Provide PyTorch native solutions for optimization techniques for transformer models, offering composability with torch.compile and distributed 
+    * For example, for QLoRA for new dtypes support
+* Interoperability: Integration with other major transformer accel libraries in PyTorch enabling smooth user journey and accelerating visibility and adoption
```
Comment (msaroufim) [inline README.md:52]:
```suggestion
* Interoperability: Work with the rest of the ecosystem such as torchtune, gpt-fast and ExecuTorch
```

**Review by msaroufim — APPROVED** [review]
stamping since the update is urgent but hopefully the feedback is still useful

File: README.md (hunk 6 above)
Comment (supriyar) [reply README.md:16]: I think they all work with both. @HDCharles or @jerryzh168 can you confirm?

**Pull request merged**